src/backend/commands/matview.c

```diff
@@ -3291,23 +3292,16 @@ clean_up_IVM_hash_entry(MV_TriggerHashEntry *entry, bool is_abort)
 	{
 		MV_TriggerTable *table = (MV_TriggerTable *) lfirst(lc);
 
-		if (table->old_tuplestores)
-		{
-			list_free(table->old_tuplestores);
-			table->old_tuplestores = NIL;
-		}
-		if (table->new_tuplestores)
+		list_free(table->old_tuplestores);
```

Comment:
here not set table->old_tuplestores to NIL, will double free ?

Reply:
list_free function behaves fine if called twice (no double-free). 
However, I can revert this change in objection that it is not related to issue I want to fix. 
I just borrowed new version clean_up_IVM_hash_entry function from Yugo NAGATA v34 IVM patch.